src/exceptions.rs

```diff
@@ -33,13 +33,50 @@ macro_rules! impl_exception_boilerplate_bound {
             ///
             /// [`PyErr`]: https://docs.rs/pyo3/latest/pyo3/struct.PyErr.html "PyErr in pyo3"
             #[inline]
-            #[allow(dead_code)]
+            #[allow(dead_code, deprecated)]
+            #[deprecated(
+                since = "0.23.0",
+                note = "Use `new_err_empty`, `new_err_arg` or `new_err_args` instead"
+            )]
             pub fn new_err<A>(args: A) -> $crate::PyErr
             where
                 A: $crate::PyErrArguments + ::std::marker::Send + ::std::marker::Sync + 'static,
             {
                 $crate::PyErr::new::<$name, A>(args)
             }
+            /// Creates a new [`PyErr`] of this type with no arguments.
+            ///
+            /// [`PyErr`]: https://docs.rs/pyo3/latest/pyo3/struct.PyErr.html "PyErr in pyo3"
+            #[inline]
+            #[allow(dead_code)]
+            pub fn new_err_empty() -> $crate::PyErr {
+                $crate::PyErr::new_empty::<$name>()
+            }
```

Comment:
It always felt weird to me that the function was e.g. `PyValueError::new_err() -> PyErr` and there is no obvious way to immediately create a `ValueError`.

I wonder, should we have e.g. `PyValueError::new() -> Bound<'py, PyValueError>` which can then be cast `.into()` PyErr? TBH, that might not be great either, having the `.into()` calls would feel quite clumsy.

Is there use for a macro here? e.g. `return Err(py_err!(PyValueError(1)))` or `return py_err!(PyValueError(1, 2, 3))`? That way we could automatically select the right constructor function according to the number of arguments.

All food for thought 🤔

Reply:
Could the `into()` directly give a `Result<T, PyErr>`? That might be ergonomic enough for many use cases.

As for the macro, I'm not sure it provides enough convenience to justify itself. Maybe yes if it reused the more general framework of a call macro?